test/unit/eigensolver/test_eigensolver.cpp

```diff
@@ -81,7 +87,14 @@ void testEigensolver(const blas::Uplo uplo, const SizeType m, const SizeType mb,
         return Matrix<T, Device::CPU>(LocalElementSize(m, m), block_size);
     }
     ();
-    matrix::util::set_random_hermitian(reference);
+    if (identity)
+      matrix::util::set(reference, [](const GlobalElementIndex& ij) {
+        if (ij.row() == ij.col())
+          return T{1};
+        return T{0};
+      });
```

Comment:
Given this is the identity matrix, would it be worth moving it to a re-usable function (`matrix::util::set_identity`)?